core/vm/evm.go

```diff
@@ -438,10 +477,20 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	}
 	evm.Context.Transfer(evm.StateDB, caller.Address(), address, value)
 
+	// Try to read code header if it claims to be EOF-formatted.
+	var header EOF1Header
+	if hasEIP3540(&evm.Config) && hasFormatByte(codeAndHash.code) {
```

Comment:
Maybe better to check for `isEOFCode` instead of `hasFormatByte`, then initcode like `EF` or `EFXX` where `XX != 00` will result in failed initcode execution and not in `ErrInvalidCodeFormat` as now.